src/components/ActivityLog/MapSplitView.tsx

```diff
@@ -129,8 +151,29 @@ export default function MapSplitView({
   }, [markerGroups, siteFeatures, strings, theme]);
 
   useEffect(() => {
-    fitBounds(boundingBox);
-  }, [boundingBox, fitBounds]);
+    if (projectId !== currentProjectId) {
+      setInitialMapViewState(undefined);
+      setCurrentProjectId(projectId);
+    }
+  }, [currentProjectId, projectId, setSearchParams]);
+
+  useEffect(() => {
+    if (!initialMapViewState) {
+      // If no map view state is provided, move map to features
+      fitBounds(boundingBox);
```

Comment:
If bounding box is not used anywhere else, perhaps it could be inside the `if` check so that it's not calculated when not needed